protocol/flows/src/v5/txrelay/flow.rs

```diff
@@ -0,0 +1,278 @@
+    pb::{kaspad_message::Payload, RequestTransactionsMessage, TransactionNotFoundMessage},
+    IncomingRoute, Router,
+};
+use std::{collections::VecDeque, sync::Arc};
+use tokio::time::timeout;
+
+pub type RelayInvMessage = Vec<TransactionId>;
+
+/// Encapsulates an incoming invs route which also receives data locally
+pub struct TwoWayIncomingRoute {
+    pub incoming_route: IncomingRoute,
+    indirect_invs: VecDeque<Vec<TransactionId>>,
+}
+
+impl TwoWayIncomingRoute {
+    pub fn new(incoming_route: IncomingRoute) -> Self {
+        Self { incoming_route, indirect_invs: VecDeque::new() }
+    }
+
+    pub fn enqueue_indirect_inv(&mut self, ids: Vec<TransactionId>) {
+        self.indirect_invs.push_back(ids)
+    }
+
+    pub async fn dequeue(&mut self) -> Result<RelayInvMessage, ProtocolError> {
+        if let Some(inv) = self.indirect_invs.pop_front() {
+            Ok(inv)
+        } else {
+            let msg = dequeue!(self.incoming_route, Payload::InvTransactions)?;
+            Ok(msg.try_into()?)
+        }
+    }
+}
+
+enum Response {
+    Transaction(Transaction),
+    NotFound(TransactionId),
+}
+
+impl Response {
+    fn transaction_id(&self) -> TransactionId {
+        match self {
+            Response::Transaction(tx) => tx.id(),
+            Response::NotFound(id) => id.to_owned(),
+        }
+    }
+}
+
+/// Flow listening to InvTransactions messages, requests their corresponding transactions if they
+/// are missing, adds them to the mempool and propagates them to the rest of the network.
+pub struct RelayTransactionsFlow {
+    ctx: FlowContext,
+    router: Arc<Router>,
+    /// A route specific for invs messages
+    invs_route: TwoWayIncomingRoute,
+    /// A route for other messages such as Transaction and TransactionNotFound
+    msg_route: IncomingRoute,
+}
+
+#[async_trait::async_trait]
+impl Flow for RelayTransactionsFlow {
+    fn name(&self) -> &'static str {
+        "RELAY_TXS"
+    }
+
+    fn router(&self) -> Option<Arc<Router>> {
+        Some(self.router.clone())
+    }
+
+    async fn start(&mut self) -> Result<(), ProtocolError> {
+        self.start_impl().await
+    }
+}
+
+impl RelayTransactionsFlow {
+    #[allow(dead_code)]
+    pub fn new(ctx: FlowContext, router: Arc<Router>, incoming_route: IncomingRoute, msg_route: IncomingRoute) -> Self {
+        Self { ctx, router, invs_route: TwoWayIncomingRoute::new(incoming_route), msg_route }
+    }
+
+    async fn start_impl(&mut self) -> Result<(), ProtocolError> {
+        // trace!("Starting relay transactions flow with {}", self.router.identity());
+        loop {
+            // Loop over incoming block inv messages
+            let inv = self.invs_route.dequeue().await?;
+            // trace!("Receive an inv message from {} with {} transaction ids", self.router.identity(), inv.len());
+
+            // Transaction relay is disabled if the node is out of sync and thus not mining
+            if self.ctx.is_ibd_running() {
```

Comment:
Note that in go we do not test "IBD is running" but rather `flow.IsNearlySynced()`. See it blockrelay/flow.rs the lines 
```
                let sink_timestamp = consensus.get_sink_timestamp();
                if sink_timestamp.is_none_or(|t| unix_now() > t + self.ctx.config.expected_daa_window_duration_in_milliseconds()) {
                    continue;
                }
```
These lines are equal to `!flow.IsNearlySynced()` in go. You can consider making this logic a shared function somewhere